core/domain/html_validation_service_test.py

```diff
@@ -1248,3 +1248,10 @@ def test_no_convertion_of_non_interactive_image_tags(self):
                 html_validation_service.convert_svg_diagram_tags_to_image_tags(
                     test_case['html_content']),
                 test_case['expected_output'])
+
+    def test_fix_incorrectly_encoded_html_string(self):
```

Comment:
I still don't know what this means. If you tell someone "this test fixes incorrectly encoded html strings", that doesn't read well. It sounds like the test is doing the fixing.

Try writing a test name that starts with something like "test that ..." -- sometimes the "that" gets omitted in the final name but you might find this framing useful for coming up with good test names. Adding the "that" to your current test name results in it not making sense. Compare this with test_cloned_object_replicates_original_object, which does make sense.

Reply:
Also, I am a bit surprised that you only have one test here for this function. Surely there are other cases to test as well? Multiple invalid chars, invalid chars at beginning/end of strings, etc.

(Note that, when writing a test, you should not be basing it on the implementation. Try to look at the function from an external perspective and have your test be a comprehensive audit of what that function should be doing.)

Reply:
@seanlip PTAL!